File: apps/api/src/routes/meeting-types.ts
import { Hono } from 'hono'
import { prisma } from '@qtscout/db'
import type { AppEnv } from '../types'

export const meetingTypes = new Hono<AppEnv>()

// Public (matches the original route, which performs no session check).
meetingTypes.get('/', async (c) => {
  try {
    const meetingTypes = await prisma.meetingType.findMany({ orderBy: { name: 'asc' } })
    return c.json(meetingTypes)
  } catch (error) {
    console.error('Error fetching meeting types:', error)
    return c.json({ error: 'Failed to fetch meeting types' }, 500)
  }
})

File: api/app/routers/meeting_types.py
from typing import Annotated

from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.db import get_session
from app.models import MeetingType
from app.schemas.meeting_type import MeetingTypeOut

router = APIRouter(prefix="/meeting-types", tags=["meeting-types"])


# Public (the Hono route performs no session check).
@router.get("", response_model=list[MeetingTypeOut])
async def list_meeting_types(
    session: Annotated[AsyncSession, Depends(get_session)],
) -> list[MeetingType]:
    result = await session.scalars(select(MeetingType).order_by(MeetingType.name.asc()))
    return list(result.all())
